executor/aggfuncs/aggfunc_test.go

```diff
@@ -234,6 +234,68 @@ func distinctUpdateMemDeltaGens(srcChk *chunk.Chunk, dataType *types.FieldType)
 	return memDeltas, nil
 }
 
+func maxUpdateMemDeltaGens(srcChk *chunk.Chunk, dataType *types.FieldType) (memDeltas []int64, err error) {
+	memDeltas = make([]int64, srcChk.NumRows())
+	var memDelta int64
+	for i := 0; i < srcChk.NumRows(); i++ {
+		row := srcChk.GetRow(i)
+		if row.IsNull(0) {
+			continue
+		}
+		switch dataType.Tp {
+		case mysql.TypeString:
+			stringT := row.GetString(0)
+			memDelta = int64(len(stringT))
+		case mysql.TypeJSON:
+			jsonT := row.GetJSON(0)
+			memDelta = int64(len(jsonT.Value))
+		case mysql.TypeEnum:
+			enumT := row.GetEnum(0)
+			memDelta = int64(len(enumT.Name))
+		case mysql.TypeSet:
+			setT := row.GetSet(0)
+			memDelta = int64(len(setT.Name))
+		}
+
+		if memDeltas[0] < memDelta {
```

Comment:
Should memDelta compare with the previous value and get the difference?

Reply:
PTAL @b41sh